src/main/java/step2/LottoResult.java

```diff
@@ -0,0 +1,52 @@
+package step2;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LottoResult {
+
+    private final List<LottoRank> lottoRanks;
+
+    public LottoResult(List<LottoRank> lottoRanks) {
+        this.lottoRanks = lottoRanks;
+    }
+
+    public LottoResult(LottoRank... lottoRanks) {
+        List<LottoRank> ranks = new ArrayList<>();
+        for (LottoRank lottoRank : lottoRanks) {
+            ranks.add(lottoRank);
+        }
+        this.lottoRanks = ranks;
+    }
+
+    public double calculateProfitRate(int money) {
+        return (calculateTotalPrizeMoney() / ((double) money * 1000));
+    }
+
+    private long calculateTotalPrizeMoney() {
+        return this.lottoRanks.stream()
+                .mapToLong(LottoRank::getPrizeMoney) //todo print문이 아님에도 get함수 사용중.
+                .sum();
+    }
```

Comment:
> LottoResult.calculateTotalPrizeMoney()에서 출력 상황이 아님에도, get함수를 쓰고 있어서 고쳐야겠다고 생각했으나, 그러면 해당 함수를 호출할때, 계속 totalSum이란 변수를 넘겨서 거기서 +=를 진행해야하나, 아니면 다른 방법이 없나 고민하다가 우선 제출하게 되었습니다ㅜㅜ

고민을 굉장히 많이하셨을걸로 보입니다! 👏 👏 👏 
결론을 말씀드리자면, getter 를 쓰는게 깔끔해보이는 로직이긴 합니다!
우리가 지양해야하는건 무분별한 getter를 지양해야하는 것이지 값을 계산하기 위한 거라면, 필요한거라면 선언해서 사용해도 될것 같아요!

만약 getter를 사용하지 않고 구현한다면 이렇게 할 수 있을것 같아요

```suggestion
    private long calculateTotalPrizeMoney() {
        return this.lottoRanks.stream()
                .mapToLong(LottoRank.NO_PRIZE::addPrize)
                .sum();
    }
```

```java
    public long addPrize(LottoRank lottoRank) {
        return lottoRank.addPrize(this.prizeMoney);
    }

    public long addPrize(long prize){
        return this.prizeMoney + prize;
    }
```

NO_PRIZE는 0원이니, 0원부터 하나씩 더하는 방식입니다.
값을 꺼내는것이 아닌 내부 상태를 메세지로 보내어 처리하는 방식이에요.

하지만, 로직이 복잡해보이고 간결하지 않아 현준님이 작성하신 로직이 나은것 같습니다 😅
한번 보고 고민해봐주세요~~ (더 고도화 해주시면 감사하겠습니다 ❤️‍🔥)

Reply:
get함수는 계속 쓰게 되었네요ㅠㅜㅜ 로직 변동만 생긴 것 같아요.
기존에는 LottoResult 에서 `List<LottoRank>` 타입을 필드로 가지고 있고, calculate 시에 List를 돌면서 총 수익을 계산했다면,
지금은 `Map<LottoRank, Long>` 타입을 필드로 가지고 있고, calculate 시에는 LottoRank의 타입 수만큼 돌면서 수익을 계산하고 있어요.